[Docker] Add 'upstream' mode which uses vanilla OTel collector and sends telemetry to Elastic via OTLP (#217)
elastic/opentelemetry-demo: .github/README.md

This works by using the demo's EDOT Collector as a gateway that forwards telemetry to the start-local EDOT Collector, which then exports to Elasticsearch.

### Upstream Mode (No EDOT)

> For users who want to use the vanilla upstream OpenTelemetry Collector instead of the Elastic Distribution (EDOT), 

Reviewer: ```suggestion
For users who do not want to use the Elastic Distributions of OpenTelemetry (EDOT)
```

My understanding is that with this PR, we are also not using our Docker images with EDOT SDKs baked in. I suggest we use the same term as our public documentation for the whole ecosystem. https://www.elastic.co/docs/reference/opentelemetry

Author: Correct, here we are not using the EDOT collector and EDOT SDKs, we are using vanilla OTel collector and SDKs. I have updated the text as per the suggestion.

while still sending telemetry to Elastic:

1. Sign up for a free trial on [Elastic Cloud](https://cloud.elastic.co/) and depending on the deployment type choose the following:
    - Elastic Cloud Hosted (ECH): In the "solution view" select "Elastic for Observability". Once that builds select Add data then Application and finally OpenTelemetry.
    - Serverless: In the "choose type" choose the "Elastic for Observability" type. Once that builds select Add data then Application and finally OpenTelemetry.
2. Copy the OTEL_EXPORTER_OTLP_ENDPOINT URL.
3. Click "Create an API Key" to create one.
4. Run the demo in upstream mode:
   ```bash
   ./demo.sh docker upstream
   ```
5. Access the demo at `http://localhost:8080`

This mode uses the standard OpenTelemetry Collector contrib image with OTLP HTTP export configured for Elastic, 

> rather than the EDOT collector. All telemetry (traces, metrics, logs) is routed to Elastic via OTLP.

Reviewer: We can also specify that no EDOT SDKs is used

Author: Yup, done.
